asserts/ifacedecls_test.go

```diff
@@ -838,6 +897,65 @@ func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsOnClassic(c
 	c.Check(rule.AllowConnection[0].OnClassic, DeepEquals, &asserts.OnClassicConstraint{Classic: true, SystemIDs: []string{"ubuntu", "debian"}})
 }
 
+func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsDeviceScope(c *C) {
+	m, err := asserts.ParseHeaders([]byte(`iface:
+  allow-connection: true`))
+	c.Assert(err, IsNil)
+
+	rule, err := asserts.CompilePlugRule("iface", m["iface"].(map[string]interface{}))
+	c.Assert(err, IsNil)
+
+	c.Check(rule.AllowInstallation[0].DeviceScope, IsNil)
+
+	tests := []struct {
+		rule     string
+		expected asserts.DeviceScopeConstraint
+	}{
+		{`iface:
+  allow-connection:
+    on-store:
+      - my-store`, asserts.DeviceScopeConstraint{Store: []string{"my-store"}}},
+		{`iface:
+  allow-connection:
+    on-store:
+      - my-store
+      - other-store`, asserts.DeviceScopeConstraint{Store: []string{"my-store", "other-store"}}},
+		{`iface:
+  allow-connection:
+    on-brand:
+      - my-brand
+      - s9zGdwb16ysLeRW6nRivwZS5r9puP8JT`, asserts.DeviceScopeConstraint{Brand: []string{"my-brand", "s9zGdwb16ysLeRW6nRivwZS5r9puP8JT"}}},
+		{`iface:
+  allow-connection:
+    on-model:
+      - my-brand/bar
+      - s9zGdwb16ysLeRW6nRivwZS5r9puP8JT/baz`, asserts.DeviceScopeConstraint{Model: []string{"my-brand/bar", "s9zGdwb16ysLeRW6nRivwZS5r9puP8JT/baz"}}},
+		{`iface:
+  allow-connection:
+    on-store:
+      - store1
+      - store2
+    on-brand:
+      - my-brand
+    on-model:
+      - my-brand/bar
+      - s9zGdwb16ysLeRW6nRivwZS5r9puP8JT/baz`, asserts.DeviceScopeConstraint{
+			Store: []string{"store1", "store2"},
+			Brand: []string{"my-brand"},
+			Model: []string{"my-brand/bar", "s9zGdwb16ysLeRW6nRivwZS5r9puP8JT/baz"}}},
+	}
```

Comment:
it would highlight some of the symmetries between plugs/slots otoh it hasn't been the style followed so far so it would be inconsistent with the preexisting tests, if you feel strongly it could be done in a follow up also touching old tests as well where it's an option

Reply:
I don't feel strongly about it. Not a blocker.